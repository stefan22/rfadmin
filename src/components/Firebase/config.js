import fb from "../../fb";

// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: fb.API_KEY,
  authDomain: fb.AUTH_DOMAIN,
  projectId: fb.PROJECT_ID,
  storageBucket: fb.STORAGE_BUCKET,
  messagingSenderId: fb.MESSAGING_SENDER_ID,
  appId: fb.APP_ID,
  measurementId: fb.MEASUREMENT_ID,
};


const prodConfig = {
  apiKey: fb.API_KEY,
  authDomain: fb.AUTH_DOMAIN,
  databaseURL: fb.DATABASE_URL,
  projectId: fb.PROJECT_ID,
  storageBucket: fb.STORAGE_BUCKET,
  messagingSenderId: fb.MESSAGING_SENDER_ID,
  appId: fb.APP_ID,
  measurementId: fb.MEASUREMENT_ID,
};

export default process.env.NODE_ENV === "production"
  ? prodConfig
  : devConfig;



