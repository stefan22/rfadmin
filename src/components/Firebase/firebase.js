import app from "firebase/app";
import "firebase/auth";
import config from "./config";

class Firebase {
  constructor() {
    app.initializeApp(config);
    this.auth = app.auth();
  }

  // auth signup
  doCreateUserWithEmailAndPassword = (email, password) =>
    this.auth.createUserWithEmailAndPassword(email, password);

  // auth signin
  doSignInWithEmailAndPassword = (email, password) =>
    this.auth.signInWithEmailAndPassword(email, password);

  // auth signout
  doSignOut = () => this.auth.signOut();

  // auth password reset
  doPasswordReset = email => this.auth.sendPasswordResetEmail(email);

  // auth password update
  doPasswordUpdate = password =>
    this.auth.currentUser.updatePassword(password);
}

export default Firebase;
