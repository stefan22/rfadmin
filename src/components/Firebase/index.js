import FirebaseContext from "./context";
import Firebase from "./firebase";
import { withFirebase } from "./context";

export default Firebase;

export { FirebaseContext, withFirebase };
