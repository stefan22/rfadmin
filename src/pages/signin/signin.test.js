import React from "react";
import { SigninForm } from "./";
import Firebase from "../../components/Firebase";

const mockUser = (
  user = {
    email: "joe@yahoo.com",
    password: "joe123",
  },
) => user;

describe("Signin comp", () => {
  it("matches snapshot when no data entered", () => {
    const wrapper = shallow(<SigninForm />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("form", () => {
    let props, wrapper;
    beforeEach(() => {
      props = {
        handleSubmit: () => {},
      };
      wrapper = shallow(<SigninForm {...props} />);
    });

    it("should have a <form /> element", () => {
      expect(wrapper.find("form").length).toBe(1);
    });
    it("should have a <form /> element with an onSubmit attribute", () => {
      expect(wrapper.find("form").props().onSubmit).toBeDefined();
    });
    it("should have onSubmit attribute of type `function`", () => {
      let form = wrapper.find("form");
      expect(typeof form.props().onSubmit === "function").toBe(true);
    });
    it("should have a user <label /> element and an <input /> element within form ", () => {
      let form = wrapper.find("form");
      expect(form.childAt(0).type()).toBe("label");
      expect(form.childAt(1).type()).toBe("input");
    });
    it("should have an <input />field with props of email equal to 'email', type:'email', placeholder: 'Email'", () => {
      let input = wrapper.find("form").childAt(1).props();
      expect(input.name).toBe("email");
      expect(input.placeholder).toBe("Email");
      expect(input.type).toBe("email");
    });

    it("should have <input /> element with onChange attributes attached", () => {
      let input = wrapper.find("form").childAt(1).props();
      expect(input.onChange).toBeDefined();
    });

    it("should submit form with the right prop values", () => {
      const user = mockUser();
      let mockSubmit = jest.fn();
      const wrapper = mount(<SigninForm onSubmit={mockSubmit} user={user} />);
      const form = wrapper.find("form");
      mockSubmit(user);
      wrapper.simulate("submit", { preventDefault: () => {} });
      expect(mockSubmit).toHaveBeenLastCalledWith({
        email: user.email,
        password: user.password,
      });
      expect(wrapper.state(user)).toEqual(mockSubmit(user));
    });
  });

  describe("initialisation state props type/values", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<SigninForm />);
    });

    it("should have an email state prop of string/empty", () => {
      expect(wrapper.state("email")).toBe("");
    });
    it("should have a password state prop of string/empty", () => {
      expect(wrapper.state("password")).toBe("");
    });

    it("should have isAdmin prop of boolean/false", () => {
      expect(wrapper.state("isAdmin")).toBe(false);
    });
    it("should have error prop set to null", () => {
      expect(wrapper.state("error")).toBe(null);
    });
  });

   describe("Firebase", () => {
    let auth, mockSubmit, wrapper;
    beforeAll(async () => {
      auth = new Firebase();
      await auth;
    });
    beforeEach(async () => {
      await auth.doSignOut();
    });

      it("should throw an Error when 'doSignInWithEmailAndPassword' with wrong password", async () => {
        let error = "";
        try {
          await auth.doSignInWithEmailAndPassword(
            "mario2224@gmail.com",
            "rabbit123",
          );
        } catch (err) {
          error = err.toString();
        }
        expect(error).toEqual(
          "Error: The password is invalid or the user does not have a password.",
        );
        auth.doSignOut();
      });

    it("should login when 'doSignInWithEmailAndPassword' with right credentials", async () => {
       
      let user = await auth.doSignInWithEmailAndPassword(
        "mario2224@gmail.com",
            "123456",
      );

      expect(user.user).toBeTruthy();
      expect(user.user.email).toEqual("mario2224@gmail.com");
    });



  }); //des Firebase



});
