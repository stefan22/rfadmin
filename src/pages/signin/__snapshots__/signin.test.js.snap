// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin comp matches snapshot when no data entered 1`] = `
<section
  className="dp-signin"
>
  <div
    className="signin-container"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <WithStyles(ForwardRef(Button))
        color="default"
        disabled={true}
        type="submit"
        variant="contained"
      >
        Signin
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
  <hr />
  <section
    className="social-wrapper"
  >
    <div
      className="thirdparty-providers"
    >
      <h4>
        OR
      </h4>
      <h5>
        Sign in with your Google Account
      </h5>
      <withRouter() />
    </div>
  </section>
</section>
`;
